### PR TITLE
hareThirdParty.hare-json: unstable-2023-09-21 -> unstable-2023-03-13

### PR DESCRIPTION
**Pull request by onemoresuza**
## Description of changes
- hareThirdParty.hare-json: move to `pkgs/development/hare-third-party`
- hareThirdParty.hare-json: refactor `configurePhase` into `makeFlags`
- hareThirdParty.hare-json: unstable-2023-09-21 -> unstable-2023-03-13

Ready `hareThirdParty.hare-json` to changes to be brought by [#271476](https://github.com/NixOS/nixpkgs/pull/271476).

## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [X] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/hare-third-party/hare-json/default.nix [hunk 1]
```diff
@@ -13,18 +13,10 @@ stdenv.mkDerivation (finalAttrs: {
...
-    export BINOUT="$NIX_BUILD_TOP/.bin"
-
-    makeFlagsArray+=(
-      PREFIX="${builtins.placeholder "out"}"
-    )
-
-    runHook postConfigure
-  '';
+  makeFlags = [
+    "HARECACHE=.cache"
```
Comment (starzation) [inline pkgs/development/hare-third-party/hare-json/default.nix:17]: Maybe keep it `.harecache`

```suggestion
    "HARECACHE=.harecache"
```

**Review by starzation — CHANGES_REQUESTED** [review]
Please keep commit message with this prefix `hareThirdParty.hare-json:`

File: pkgs/development/hare-third-party/hare-json/default.nix (hunk 1 above)
Comment (onemoresuza) [reply pkgs/development/hare-third-party/hare-json/default.nix]: Done it.

**Comment by onemoresuza** [comment]
> Please keep commit message with this prefix `hareThirdParty.hare-json:`

Thanks catching my mistake. Rebased the commit messages to start with `hareThirdParty.hare-json`.

**Review by starzation — APPROVED** [review]

**Comment by pbsds** [comment]
#272136 is merged, please rebase

**Comment by onemoresuza** [comment]
Rebase against `master` after [#271476](https://github.com/NixOS/nixpkgs/pull/271476).

I've left the file location change as a separate commit, based on what is done
when changing a package from the [`Category Hierarchy`][category-hierarchy] to
`pkgs/by-name`. Is this correct or should such commit be squashed?

[category-hierarchy]: <https://github.com/NixOS/nixpkgs/tree/master/pkgs#category-hierarchy>

**Comment by pbsds** [comment]
Result of `nixpkgs-review pr 272136` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>hareThirdParty.hare-json</li>
  </ul>
</details>


LGTM, thanks :)

**Comment by pbsds** [comment]
> Is this correct or should such commit be squashed?

In general, splitting each change into separate commits makes review far easier. Github/git is usually also better able to create meaningful diff views when each change is smaller. I know however there are other committers who prefer the opposite.

**Pull request merged**